### PR TITLE
Adding subcommand completions for launchctl

### PR DESCRIPTION
**Pull request by samdmarshall**
## Description

Adding subcommand completions for `launchctl` in the same manner that `gem` has. I don't know if additional things need to be done so this doesn't get populated for systems that don't have `launchctl` (non-os x systems). 

Is there a way to auto-generate subcommand completions for subcommands? It only seemed to work for command line flags (`-` or `--`).

## TODOs:
- [x] Changes to fish usage are reflected in user documenation/manpages.
- [x] Tests have been added for regressions fixed

### CODE REVIEW

**Comment by faho** [comment]
>I don't know if additional things need to be done so this doesn't get populated for systems that don't have launchctl (non-os x systems).

Completions are autoloaded, so there's no issue there. The worst that'll happen is that people get completions for a command they don't have.

>Is there a way to auto-generate subcommand completions for subcommands? It only seemed to work for command line flags (- or --).

Unfortunately no, because there's no real standard parsable format for how they should be mentioned in man pages.

I see this adds subcommands via something like `xa 'bootstrap\t"'(_ "Bootstraps a domain or a service into a domain")'"'`. This isn't too pretty, but probably the best way to offer localization. That's something we should probably take more seriously in general (heck, I should go and check the german one).

`__fish_use_subcommand` is a very simple function - it assumes any bare (i.e. not starting with "-") token is a subcommand.  From [the man page](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man1/launchctl.1.html), this seems to be okay, since there are no options whatsoever to launchctl in general (and especially none that take arguments, which would break with `__fish_use_subcommand`.

Long story short: Merged, thanks!

**Comment by samdmarshall** [comment]
Compared to the "regular" flag-based completions, these seem to be a bit of a hack/magic to get working. Any chance that this could be addressed as an enhancement to the `complete` function to add proper support for subcommands (as many Ruby and Python based tools use subcommands rather than flags)?

**Comment by faho** [comment]
The long-term plan is #478 - replacing the current completion system with something based on docopt. From https://github.com/fish-shell/fish-shell/issues/478#issuecomment-126814563 and @ridiculousfish's following comments, the syntax would be something like

```
complete --signature '
    Usage:
        ip link show <device>
        ip addr [add | del] <ifaddr> dev <devname>
        ip route [ add | del | change | append | replace | monitor ] <route>
'
```
